### PR TITLE
Play Client Interpreter

### PR DESCRIPTION
**Pull request by gaeljw**
This PR aims to solve #738.

Things to note from a user perspective:
- multipart requests are not supported: this would require copying a bunch of stuff provided in Play framework that are not provided in Play Standalone WS (see https://github.com/playframework/play-ws/issues/126)
- streaming only works with `AkkaStreams` capability

Things to note for the review:
- I had to slighty change some tests that were relying on the output order of header values because Play client doesn't preserve the order (I didn't ready the HTTP spec but I don't think this is a requirement)
- I moved the multipart tests to a separate class so that _sttp-client_ can implement them but not necessarily _play-client_
- I moved the documentation for clients in a sub directory

All feedbacks are welcomed :)

### CODE REVIEW

File: doc/client/play.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- a description of the request to be made, with the input value
+  encoded as appropriate request parameters: path, query, headers and body.
+  This can be further customised and sent using regular Play methods.
+- a response parser to be applied to the response got after executing the request.
+  The result will then contain the decoded error or success values
+  (note that this can be the body enriched with data from headers/status code).
+
+Example:
+
+```scala
```
Comment (adamw) [inline doc/client/play.md:42]: I'd try adding `scala mdoc:compile-only` to that this example is compiled (when running `mdoc` in the documentation project)

File: client/tests/src/main/scala/sttp/tapir/client/tests/ClientTests.scala [hunk 2]
```diff
@@ -52,8 +52,8 @@ abstract class ClientTests[R] extends AnyFunSuite with Matchers with PortCounter
     case r @ GET -> Root / "api" / "echo" / "params"                       => Ok(r.uri.query.params.toSeq.sortBy(_._1).map(p => s"${p._1}=${p._2}").mkString("&"))
     case r @ GET -> Root / "api" / "echo" / "headers" =>
       val headers = r.headers.toList.map(h => Header(h.name.value, h.value.reverse))
-      val filteredHeaders = r.headers.find(_.name.value == "Cookie") match {
-        case Some(c) => headers.filter(_.name.value == "Cookie") :+ Header("Set-Cookie", c.value.reverse)
+      val filteredHeaders = r.headers.find(_.name.value.equalsIgnoreCase("Cookie")) match {
```
Comment (adamw) [inline client/tests/src/main/scala/sttp/tapir/client/tests/ClientTests.scala:55]: 👍

**Comment by adamw** [comment]
Thanks, looks really solid!

I suppose the multipart tests need to be disabled? Otherwise it's good to merge :)

**Comment by adamw** [comment]
Ah they are disabled 🤦 nevermind ;)

**Comment by adamw** [comment]
Merging then :)

File: doc/client/play.md (hunk 1 above)
Comment (gaeljw) [reply doc/client/play.md:42]: Sure, I'm not familiar with `mdoc` but it's an opportunity to see how it works :)

I'll do it in another PR as I see you've already merged this one.

**Comment by gaeljw** [comment]
Thanks for the quick review :+1:

File: doc/client/play.md (hunk 1 above)
Comment (adamw) [reply doc/client/play.md:42]: I've fixe dthis already :)

Comment (adamw) [reply doc/client/play.md:42]: https://github.com/softwaremill/tapir/commit/4d56e0ef26aac2a6099e0b16a121ebd6c2ffdbae

**Comment by gaeljw** [comment]
@adamw Not sure, but I might have broke the documentation for the clients. I don't see the links to the clients pages at https://tapir.softwaremill.com/ (https://tapir.softwaremill.com/en/latest/sttp.html still accessible though).

**Comment by adamw** [comment]
@gaeljw Fixed, two of the generated doc files where missing

File: doc/client/play.md (hunk 1 above)
Comment (gaeljw) [reply doc/client/play.md:42]: Oh great thanks :)

**Comment by adamw** [comment]
Released in https://github.com/softwaremill/tapir/releases/tag/v0.17.0-M7, thanks!

**Pull request merged**